Week3/homework/step2-6.js

```diff
@@ -2,15 +2,28 @@
 
 const arr2d = [[1, 2], [3, 4], [5, 6]];
 const arr3d = [[[1, 2], [3, 4]], [[5, 6], [7, 8]]];
+const flattenedArr2d = [];
+const flattenedArr3d = [];
 
 function flattenArray2d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let i = 0; i < arr.length; i++) {
+    for (let j = 0; j < arr[i].length; j++) {
+      flattenedArr2d.push(arr[i][j]);
+    }
+  }
+  return flattenedArr2d;
 }
 
 function flattenArray3d(arr) {
-  // Replace this comment and the next line with your code
-  console.log(arr);
+  for (let k = 0; k < arr.length; k++) {
+    for (let l = 0; l < arr[k].length; l++) {
+      for (let m = 0; m < arr[k][l].length; m++) {
+        flattenedArr3d.push(arr[k][l][m]);
+      }
+    }
+  }
+
+  return flattenedArr3d;
```

Comment:
Logic works fine but imagine if you have to do this for a 4-D or a 5-D array. What happens in that case? Suggestion: try reading about a programming concept called recursion and see if you can apply the same here. Also, look for `Array.prototype.flat` method on MDN. 🙂

Reply:
I will look flat method. 👍

Reply:
I will learn flat concept